core/registry.js

```diff
@@ -220,6 +231,10 @@ Blockly.registry.getClassFromOptions = function(type, options) {
   var typeName = type.toString();
   var plugin = options.plugins[typeName] || Blockly.registry.DEFAULT;
 
+  if (Blockly.registry.flyoutTypes_.indexOf(type) > -1 && options.plugins.flyouts) {
+    plugin = options.plugins.flyouts[typeName] || Blockly.registry.DEFAULT;
```

Comment:
Not a huge fan of this approach, as it means whenever we want to add another plugin group, we'd need to add additional login in the global registry to support that group.
Here are a couple of alternatives: 

Only support a single flat list in plugins, for semantically similar groups, you can have a shared prefix, eg: 
```
{
   ...
   'plugins': {
         'toolbox': ...,
         'flyouts-toolbox': ...,
         'flyouts-trashcan': ...,
         'flyouts-mutator': ...,
    }
}
```

Alternately, you can support a nested list in plugins, but flatten it when parsed. such that this:
```
{
    'plugins': {
          'flyouts': {
               'toolbox': ...,
               'trashcan': ...,
           }
     }
}
```
turns into: 
```
{
     'plugins': {
         'flyouts.toolbox': ...,
         'flyouts.trashcan': ...,
     }
}
```

Reply:
ok, went with option 1 for this pr. If we decide we really want a nested structure I can update in the future.